tests/sentry/integrations/slack/test_utils.py

```diff
@@ -73,6 +99,27 @@ def test_valid_private_channel_selected(self):
         )
         self.run_valid_test("#my-private-channel", CHANNEL_PREFIX, "m-p-c", False)
 
+    @with_feature("organizations:slack-sdk-get-channel-id")
+    @patch("sentry.integrations.slack.sdk_client.metrics")
+    @patch("slack_sdk.web.client.WebClient._perform_urllib_http_request")
+    def test_valid_private_channel_selected_sdk(self, mock_api_call, mock_metrics):
+        # Tests chat_scheduleMessage and chat_deleteScheduledMessage
+        mock_api_call.return_value = {
+            "body": orjson.dumps(
+                {"ok": True, "channel": "m-p-c", "scheduled_message_id": "Q1298393284"}
+            ).decode(),
+            "headers": {},
+            "status": 200,
+        }
+
+        self.run_valid_test("#my-private-channel", CHANNEL_PREFIX, "m-p-c", False)
+
+        mock_metrics.incr.assert_called_with(
+            SLACK_DATADOG_METRIC,
+            sample_rate=1.0,
+            tags={"ok": True, "status": 200},
+        )
+
     def test_valid_member_selected(self):
```

Comment:
can we test this with the sdk? i don't think the current logic will correctly search for a user if a channel does not exist with the id